### PR TITLE
Update GetArgument<T> to use Argument<T>

### PR DESCRIPTION
**Pull request by joemcbride**
Github: #74 #109

### CODE REVIEW

**Comment by joemcbride** [comment]
@tlil87 Can you tell me if this is what you envision as expected behavior for `GetArgument<T>` when an "overriding" default value is provided?  I'm guessing not ...

The overriding default value will only be used if there wasn't a value found in `Arguments`.  So, if a default value was provided during the Argument creation, then the overriding default value would never be used.

File: src/GraphQL.Tests/Builders/FieldBuilderTests.cs [hunk 1]
```diff
@@ -161,47 +162,116 @@ public void getting_unspecified_argument_in_resolver_yields_null()
...
+                    context.GetArgument<int?>("skip").ShouldEqual(null);
+                    return null;
+                });
+
+            var field = objectType.Fields.First();
+            field.Resolve(new ResolveFieldContext
+            {
+                Arguments = new Dictionary<string, object>
+                {
+                    { "skip", null }
```
Comment (tlil87) [inline src/GraphQL.Tests/Builders/FieldBuilderTests.cs]: This test would still pass without `skip => null` being passed in here, yeah?

**Comment by tlil87** [comment]
@joemcbride, this looks good and in line with what I was envisioning :+1: Thanks for fixing.

File: src/GraphQL.Tests/Builders/FieldBuilderTests.cs (hunk 1 above)
Comment (joemcbride) [inline src/GraphQL.Tests/Builders/FieldBuilderTests.cs]: It would, yes.  I'll add a test for that too.

**Pull request merged**